sp-sim/Cargo.toml

```diff
@@ -4,6 +4,10 @@ version = "0.1.0"
 edition = "2021"
 license = "MPL-2.0"
 
+[features]
+old-state = []
+no-caboose = []
```

Comment:
Do these need to be cargo features? I worry that they won't get tested as thoroughly or used as frequently if they aren't the default.

Reply:
These are intended for testing some of the backwards compatibility state we need to support.

Reply:
Could they be runtime / config parameters instead? (I'm thinking about something like a4x2, where changing a cargo feature requires a rebuild + redeploy, but changing a config file could be done without that lap.)

Reply:
Good suggestion, I converted them over to take an option in the config file